### ISSUE TITLE
InconsistentMroError exception shown as traceback

### ISSUE THREAD

**Issue by andy-maier** [PyCQA/astroid#822]
### Steps to reproduce

1. Have the same mixin class (RegexpMixin) twice in the inheritance tree, like this (using the class names from the exception below), causing an MRO conflict:
   - class TestSubscriptionsClass (RegexpMixin, PyWBEMServerClass)
   - class RegexpMixin (object)
   - class PyWBEMServerClass (RegexpMixin, PegasusServerTestBase)
   - class PegasusServerTestBase (ClientTest)
   - class ClientTest (unittest.TestCase)
   - class unittest.TestCase (object)
2. Run pylint

I have created https://github.com/pywbem/pywbem/pull/2369 from this version of the code to demonstrate it. Travis run on Python 3.8: https://travis-ci.org/github/pywbem/pywbem/jobs/712077759#L1987

### Current behavior

Pylint/astroid correctly recognizes the MRO resolution conflict (RegexpMixin being both before and after PyWBEMServerClass). However, in the course of detecting and reporting that, it shows an exception traceback that starts with a KeyError in astroid/decorators.py line 31 because the func is not yet cached, and Python then reports the InconsistentMroError in astroid/scoped_nodes.py line 77 as a second exception while handling the first one. It seems to recover from that because it gets to completion and pylint reports the finding as F0002, although with incorrect line/col.

The issue is that the traceback is confusing to users.
Possibly (did not verify), the incorrect line/col reported by Pylint is also caused by this.

```
pylint --disable=fixme --rcfile=pylintrc ...(long file list)...
Traceback (most recent call last):
  File "/Users/maiera/virtualenvs/pywbem38/lib/python3.8/site-packages/astroid/decorators.py", line 31, in cached
    return cache[func]
KeyError: <bound method ClassDef.slots of <ClassDef.TestSubscriptionsClass l.5878 at 0x116b90be0>>

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/maiera/virtualenvs/pywbem38/lib/python3.8/site-packages/pylint/lint.py", line 1212, in get_ast
    return MANAGER.ast_from_file(filepath, modname, source=True)
  File "/Users/maiera/virtualenvs/pywbem38/lib/python3.8/site-packages/astroid/manager.py", line 95, in ast_from_file
    return AstroidBuilder(self).file_build(filepath, modname)
  File "/Users/maiera/virtualenvs/pywbem38/lib/python3.8/site-packages/astroid/builder.py", line 136, in file_build
    return self._post_build(module, encoding)
  File "/Users/maiera/virtualenvs/pywbem38/lib/python3.8/site-packages/astroid/builder.py", line 156, in _post_build
    self.delayed_assattr(delayed)
  File "/Users/maiera/virtualenvs/pywbem38/lib/python3.8/site-packages/astroid/builder.py", line 230, in delayed_assattr
    if not _can_assign_attr(inferred, node.attrname):
  File "/Users/maiera/virtualenvs/pywbem38/lib/python3.8/site-packages/astroid/builder.py", line 57, in _can_assign_attr
    slots = node.slots()
  File "/Users/maiera/virtualenvs/pywbem38/lib/python3.8/site-packages/astroid/decorators.py", line 33, in cached
    cache[func] = result = func(*args, **kwargs)
  File "/Users/maiera/virtualenvs/pywbem38/lib/python3.8/site-packages/astroid/scoped_nodes.py", line 2742, in slots
    slots = list(grouped_slots())
  File "/Users/maiera/virtualenvs/pywbem38/lib/python3.8/site-packages/astroid/scoped_nodes.py", line 2727, in grouped_slots
    for cls in self.mro()[:-1]:
  File "/Users/maiera/virtualenvs/pywbem38/lib/python3.8/site-packages/astroid/scoped_nodes.py", line 2813, in mro
    return self._compute_mro(context=context)
  File "/Users/maiera/virtualenvs/pywbem38/lib/python3.8/site-packages/astroid/scoped_nodes.py", line 2803, in _compute_mro
    return _c3_merge(unmerged_mro, self, context)
  File "/Users/maiera/virtualenvs/pywbem38/lib/python3.8/site-packages/astroid/scoped_nodes.py", line 77, in _c3_merge
    raise exceptions.InconsistentMroError(
astroid.exceptions.InconsistentMroError: Cannot create a consistent method resolution order for MROs (RegexpMixin, object), (PyWBEMServerClass, RegexpMixin, PegasusServerTestBase, ClientTest, TestCase, object), (RegexpMixin, PyWBEMServerClass) of class <ClassDef.TestSubscriptionsClass l.5878 at 0x116b90be0>.
************* Module tests.manualtest.run_cim_operations
tests/manualtest/run_cim_operations.py:1:0: F0002: <class 'astroid.exceptions.InconsistentMroError'>: Cannot create a consistent method resolution order for MROs (RegexpMixin, object), (PyWBEMServerClass, RegexpMixin, PegasusServerTestBase, ClientTest, TestCase, object), (RegexpMixin, PyWBEMServerClass) of class <ClassDef.TestSubscriptionsClass l.5878 at 0x116b90be0>. (astroid-error)


Report
======
```

### Expected behavior

No traceback in this case.

One way to avoid the traceback is to reset the exception context of the active KeyError exception before calling func in astroid/decorators.py line 31:

```
    try:
        return cache[func]
    except KeyError as exc:
        exc.__cause__ = None  # Avoid 'During handling of the above exception...'
        cache[func] = result = func(*args, **kwargs)
        return result
```

I did not verify that for this case, though.

See [PEP 415](https://www.python.org/dev/peps/pep-0415/)

###

pylint --version
pylint 2.4.4
astroid 2.3.3
Python 3.8.5 (default, Jul 21 2020, 10:48:26) 
[Clang 11.0.3 (clang-1103.0.32.62)]

**Comment by PCManticore** [comment]
Thanks for the clear example and for the report.

**Comment by degustaf** [comment]
I believe this is another symptom of the bug fixed in #1089, and should be closed.

**Comment by Pierre-Sassoulas** [comment]
Thank you @degustaf, cleaning up old issues is really hard, I appreciate the help.
